File: src/1979.find-greatest-common-divisor-of-array.2106/solution.ts
function findGCD (nums: number[]): number {
    let max = -Infinity;
    let min = Infinity;
    for (let i = 0; i < nums.length; i++) {
        max = Math.max(nums[i], max);
        min = Math.min(nums[i], min);
    }
    return gcd(max, min);
}

function gcd (a:number, b:number) {
    while (b !== 0) {
        const tmp = a % b;
        a = b;
        b = tmp;
    }
    return a;
}

File: src/1979.find-greatest-common-divisor-of-array.2106/solution.java
class Solution {
    public int findGCD(int[] nums) {
        int max = nums[0];
        int min = nums[0];
        for(int i=1;i<nums.length;i++){
            max = Math.max(max,nums[i]);
            min = Math.min(min,nums[i]);
        }
        return gcd(max,min);
    }
    
    private int gcd(int a,int b){
        if(b == 0){
            return a;
        }
        return gcd(b,a%b);
    }
}